File: problems/leetcode/242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int c[26] = {0};
        for (char x : s) c[x - 'a']++;
        for (char x : t) c[x - 'a']--;
        for (int v : c) {
            if (v != 0) return false;
        }
        return true;
    }
};

File: problems/leetcode/242.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        letters_s = {}
        letters_t = {}
        for i in range(len(s)):
            letters_s[s[i]] = letters_s.get(s[i], 0) + 1
            letters_t[t[i]] = letters_t.get(t[i], 0) + 1
        return letters_s == letters_t